conftest.py

```diff
@@ -97,16 +96,27 @@ def pytest_generate_tests(metafunc):
     except ImportError:
         Display = None
 
-if not constants.SERVER_CONFIG_FS.exists(constants.SERVER_CONFIG_FILE):
-    print('\n configure testdata')
-    # ToDo check pytest tmpdir_factory
-    examples = DataFiles(data_fs=constants.DATA_FS,
-                         server_config_fs=constants.SERVER_CONFIG_FS)
-    examples.create_server_config(detailed_information=True)
-    examples.create_data()
 
-if not constants.SERVER_CONFIG_FS.exists(constants.MSCOLAB_CONFIG_FILE):
-    config_string = f'''
+def generate_initial_config():
+    """Generate an initial state for the configuration directory in tests.constants.ROOT_FS
+    """
+    if not constants.ROOT_FS.exists("msui/testdata"):
+        constants.ROOT_FS.makedirs("msui/testdata")
+
+    # make a copy for mscolab test, so that we read different pathes during parallel tests.
+    sample_path = os.path.join(os.path.dirname(__file__), "tests", "data")
+    shutil.copy(os.path.join(sample_path, "example.ftml"), constants.ROOT_DIR)
+
+    if not constants.SERVER_CONFIG_FS.exists(constants.SERVER_CONFIG_FILE):
+        print('\n configure testdata')
+        # ToDo check pytest tmpdir_factory
+        examples = DataFiles(data_fs=constants.DATA_FS,
```

Comment:
a note: this is expensive. A later improvement, may be creation once and then a copy to the destination

Reply:
> a note: this is very expensive.

It seems to add about ~3 minutes to the test duration in the serial test suite.

I tried to keep the changes relatively minor here to avoid breaking even more, until the test suite is in a somewhat sane state (i.e. passes without retries). Ideally this directory would be split up and e.g. those datafiles should become their own fixture that can then be scoped to the session.

> Once created for each used core may be enough.

This would happen automatically then, if it was a session-scoped fixture.

> Maybe we should mark tests where we need this data and create it only then.

This would happen automatically with a fixture as well, since every test that needs it would have to request the fixture (that is if we don't make it `autouse` and of course, if it is session-scoped it wouldn't really matter how many tests require it).

As it stands I think the impact is not too high on the test duration and refactoring this into a fixture can be a future PR.

Reply:
ok, it is not only the time, we also create and drop a huge amount of data which we should optimize.